src/server/address_space_internal.cpp

```diff
@@ -295,6 +295,64 @@ namespace OpcUa
       return StatusCode::BadAttributeIdInvalid;
     }
 
+    void AddressSpaceInMemory::SetMethod(const NodeId& node, std::function<std::vector<OpcUa::Variant> (std::vector<OpcUa::Variant> arguments)> callback)
+    {
+      NodesMap::iterator it = Nodes.find(node);
```

Comment:
You may have a point here.  We could have another class handling the method
call. and keep the node link with a map<NodeId, method>. But then we need
to remove map object when its node is deleted...

maybe I will test the idea python first

>